common/frame_session.go

```diff
@@ -827,17 +845,19 @@ func (fs *FrameSession) onAttachedToTarget(event *target.EventAttachedToTarget)
 	// Handle new worker
 	w, err := NewWorker(fs.ctx, session, targetID, event.TargetInfo.URL)
 	if err != nil {
-		if !fs.page.browserCtx.browser.connected && strings.Contains(err.Error(), "websocket: close 1006 (abnormal closure)") {
-			// If we're no longer connected to browser, then ignore WebSocket errors
+		// ignore errors if we're no longer connected to browser
+		// this happens when there is no browser but we still want to
+		// attach a worker to it.
+		if !fs.page.browserCtx.browser.connected {
```

Comment:
Should we use `browser.IsConnected()` here like on line 811, to make sure it's read locked?

Reply:
Good catch 🙇